python-avd/pyavd/schema/metaschema/meta_schema_model.py

```diff
@@ -0,0 +1,559 @@
+    _resolve_schema: ClassVar[bool] = True
+
+    def __init__(self, resolve_schema: bool | None = None, **data):
+        """
+        Overrides BaseModel.__init__(**data).
+        Takes a kwarg "resolve_schema" which controls if all subclasses of AvdSchemaBaseModel should expand any $ref in the input schema.
+
+        The $ref expansion _only_ covers this field.
+        Any $ref on child fields are expanded as they are initialized by Pydantic since they are based on this base class.
+        """
+
+        # Setting the resolve_schema attribute on the class, so all sub-classes will inherit this automatically.
+        if resolve_schema is not None:
+            AvdSchemaBaseModel._resolve_schema = resolve_schema
+
+        if self._resolve_schema:
+            data = merge_schema_from_ref(data)
+
+        super().__init__(**data)
+
+    @cached_property
+    def _descendant_tables(self) -> set[str]:
+        """
+        Descendant tables are used for schema docs to identify if a key should be included in a certain table.
+
+        Descendant tables should return all table names from fields below this field. Not the field itself.
+
+        There are no descendant tables on most fields so the base class returns an empty set. Overridden on list and dict.
+        """
+        return set()
+
+    @cached_property
+    def _table(self) -> str | None:
+        """
+        Return the name of the schema documentation table where this field should be included.
+
+        The table name can be derived from several sources depending on the position and configuration of the field.
+        In order the sources are:
+        - Statically defined under "documentation_options.table".
+        - Inherit from parent schema if available.
+        - None if this is the root dict.
+          This means that the first level of keys - called root keys - will not find a parent table.
+        - Root keys will default to a hyphen variant of their key name.
+
+        Most fields will inherit from parent schema.
+        """
+        if self.documentation_options and self.documentation_options.table:
+            return self.documentation_options.table
+
+        # No local table, so use the _table from the parent_schema if available.
+        if self._parent_schema and self._parent_schema._table:
+            return self._parent_schema._table
+
+        # This should never happen, since only the root key should be without a parent_schema.
+        if len(self._path) != 1:
+            raise NotImplementedError("Something went wrong in _table", self._path)
+
+        # This is a root key the default table is the key with hyphens and removing <,>
+        return self._key.replace("<", "").replace(">", "").replace("_", "-")
+
+    @cached_property
+    def _path(self) -> list[str]:
+        """
+        Returns the variable path for this field to be used in schema docs.
+        Like "rootkey.subkey.[].mykey".
```

Comment:
is this notation a standard ?

I am more used to RESTCONF like path
```suggestion
        Like "rootkey.subkey[].mykey".
```

(which granted is less easy to parse :)